Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -2473,9 +2478,19 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
                     return .success(resolved)
                 } else if reference.fragment != nil, nodeAnchorSections.keys.contains(reference) {
                     return .success(reference)
-                } else if let alias = referenceAliases[reference] {
+                } else if let alias = canonicalReferences[reference] {
                     return .success(alias)
                 } else {
+                    let parent = reference.removingLastPathComponent()
```

Comment:
There is a parent reference argument passed to this method but it's being shadowed here. 

It would be helpful to add a comment here explaining why this is using the parent of the reference being resolved as opposed to the "parent" reference argument  that serves as an enclosing search context.